packages/repository/src/relations/has-many/has-many-through-repository.factory.ts

```diff
@@ -51,27 +51,34 @@ export function createHasManyThroughRepositoryFactory<
 ): HasManyThroughRepositoryFactory<Target, TargetID, Through, ForeignKeyType> {
   const meta = resolveHasManyThroughMetadata(relationMetadata);
   const result = function (fkValue: ForeignKeyType) {
-    function getTargetContraint(
-      throughInstances: Through | Through[],
+    function getTargetConstraintOnThrough(
+      throughInstances: Through[],
     ): DataObject<Target> {
-      return createTargetConstraint<Target, Through>(meta, throughInstances);
+      return createTargetConstraintOnThrough<Target, Through>(
+        meta,
+        throughInstances,
+      );
+    }
+    function getTargetKey(throughInstances: Through[]): TargetID {
+      return getTargetKeyFromThroughModel(meta, throughInstances);
     }
-    function getThroughConstraint(): DataObject<Through> {
-      const constriant: DataObject<Through> = createThroughConstraint<
+    function getThroughConstraintOnSource(): DataObject<Through> {
+      const constraint: DataObject<Through> = createThroughConstraintOnSource<
         Through,
         ForeignKeyType
       >(meta, fkValue);
-      return constriant;
+      return constraint;
     }
 
-    function getThroughFkConstraint(
-      targetInstance: Target,
+    function getThroughConstraintOnTarget(
+      fkValues: TargetID,
```

Comment:
Is the `fkValues` an array? if so I think the type should be `TargetID[]` :p

Reply:
I think the `any` type escaped from the type checking. Converted it to `TargetID[]` in https://github.com/strongloop/loopback-next/pull/5719/commits/975f25ec03866f4b65c41d380392214735b7fceb.